src/NexusMods.App.UI/Pages/LibraryPage/LibraryViewModel.cs

```diff
@@ -505,23 +793,129 @@ await Parallel.ForAsync(
+        }
+
+        // Use the efficient method directly from ModUpdateService
+        var modPagesWithUpdates = _modUpdateService.GetAllModPagesWithUpdates();
+        
+        foreach (var (modPageId, updates) in modPagesWithUpdates)
+        {
+            // Create a minimal model for the message (we need this for future/existing infrastructure)
+            var model = new CompositeItemModel<EntityId>(modPageId);
+            var message = new UpdateAndReplaceMessage(updates, model);
+            await HandleUpdateAndReplaceMessage(message, cancellationToken);
+        }
+    }
+
+    /// <summary>
+    /// Removes old library items that are not installed in any read-only collections.
+    /// This helps keep the library clean by removing outdated versions that are only in modifiable collections.
+    /// Items that are still installed in read-only collections are preserved.
+    /// </summary>
+    /// <param name="oldToNewLibraryMapping">The mapping of old library items to their new replacements.</param>
+    private async ValueTask RemoveOldLibraryItemsIfNotInReadOnlyCollections(IEnumerable<(LibraryItem.ReadOnly oldItem, LibraryItem.ReadOnly newItem)> oldToNewLibraryMapping)
+    {
+        var oldItemsToRemove = new List<LibraryItem.ReadOnly>();
+        
+        foreach (var (oldItem, newItem) in oldToNewLibraryMapping)
+        {
+            // If the new item is not valid (e.g., download failed), we should not remove the old item.
+            if (!newItem.IsValid())
+                continue;
+            
+            // Check if the old item is still linked to any collections
+            var collectionsWithItem = _libraryService.CollectionsWithLibraryItem(oldItem, excludeReadOnlyCollections: false);
+            
+            // Only remove if the item is NOT in any read-only collections (i.e., only in modifiable collections or no collections)
+            var hasReadOnlyCollections = collectionsWithItem.Any(x => x.collection.IsReadOnly);
+            if (!hasReadOnlyCollections)
+                oldItemsToRemove.Add(oldItem);
+        }
+
+        if (oldItemsToRemove.Count > 0)
+            await _libraryService.RemoveLibraryItems(oldItemsToRemove);
+    }
 }
 
 public readonly record struct InstallMessage(LibraryItemId[] Ids);
-public readonly record struct UpdateMessage(ModUpdatesOnModPage Updates, CompositeItemModel<EntityId> TreeNode);
+public readonly record struct UpdateAndReplaceMessage(ModUpdatesOnModPage Updates, CompositeItemModel<EntityId> TreeNode);
+public readonly record struct UpdateAndKeepOldMessage(ModUpdatesOnModPage Updates, CompositeItemModel<EntityId> TreeNode);
 public readonly record struct ViewChangelogMessage(OneOf<NexusModsModPageMetadataId, NexusModsLibraryItemId> Id);
 public readonly record struct ViewModPageMessage(OneOf<NexusModsModPageMetadataId, NexusModsLibraryItemId> Id);
 public readonly record struct HideUpdatesMessage(OneOf<NexusModsModPageMetadataId, NexusModsLibraryItemId> Id);
```

Comment:
These should all take a collection of items rather than a single item. To allow for multi-selection.

In particular for the Update ones since this is being reused for the Update All button.